src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -58,15 +58,27 @@ def _get_start_and_end_indices(data: list[TimeSeriesPoint]) -> tuple[int, int]:
     return start, end
 
 
-def handle_send_historical_data_to_seer(alert_rule: AlertRule, project: Project, method: str):
+def handle_send_historical_data_to_seer(
+    alert_rule: AlertRule,
+    snuba_query: SnubaQuery,
+    project: Project,
+    method: str,
+    event_types: list[SnubaQueryEventType.EventType] | None = None,
+):
```

Comment:
it's a bit of a code smell that we need to update this methods params without really changing how it's working.

Do we need to add both the snuba query and the event types? is this method invoked / used outside of the `send_new_rule_data`?

🤔 It almost feels like this file could be a class, where we initialize it, run through a few steps / data formatting, then have a method to return the results of the query that was built in the constructor. using that design, we could DRY a lot of the code, and access member variables without having to proxy them through many layers of method invocations.

---

do have a general outline of how these methods connect together? i think we could also do this w/o making a class, but it's hard to grok the method order and each one has to guard against bad input because of how everything is being used.

Reply:
It's handling both when an alert is created and when it's edited which have different data - in the create case, we've already made the alert rule so we can just pass the entire thing in and pull off what we need (and we delete the rule entirely if something fails).  In the update case we can't risk updating _before_ sending the data to Seer in case there's some error, the alert would be in a bad state. We have to pass the updated data unattached to the rule along.

Reply:
🤔 would it be helpful to split the create / update cases? then we could have a few helper methods that add to / mutate the data, before passing it along to seer.